File: app/src/main/java/dev/suhockii/lifetest/util/ui/fragment/NonLeakFragment.java
package dev.suhockii.lifetest.util.ui.fragment;

import dev.suhockii.lifetest.util.mvp.SnackbarView;
import io.reactivex.disposables.CompositeDisposable;

public abstract class NonLeakFragment extends InteractionFragment implements SnackbarView {

    protected CompositeDisposable compositeDisposable = new CompositeDisposable();

    @Override
    public void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear();
    }
}

File: app/src/main/java/dev/suhockii/lifetest/util/ui/fragment/NonLeakFragment.kt
package dev.suhockii.lifetest.util.ui.fragment

import dev.suhockii.lifetest.util.mvp.SnackbarView
import io.reactivex.disposables.CompositeDisposable

abstract class NonLeakFragment : InteractionFragment(), SnackbarView {

    private var compositeDisposable = CompositeDisposable()

    override fun onDestroy() {
        super.onDestroy()
        compositeDisposable.clear()
    }

    fun getCompositeDisposable() = compositeDisposable
}
